app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt

```diff
@@ -131,399 +115,170 @@ class ProfileChooserFragmentTest {
-        )
-      ).check(
-        matches(
-          withText("G")
-        )
-      )
-      onView(withId(R.id.profile_recycler_view)).perform(
-        scrollToPosition<RecyclerView.ViewHolder>(
-          9
-        )
-      )
-      onView(
-        atPositionOnView(
-          R.id.profile_recycler_view,
-          9, R.id.profile_name_text
-        )
-      ).check(
-        matches(
-          withText("H")
-        )
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 0)
+      matchStringOnProfileListItem(
+        position = 0,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "Admin"
+      )
+      scrollToPosition(position = 1)
+      matchStringOnProfileListItem(
+        position = 1,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "A"
+      )
+      scrollToPosition(position = 2)
+      matchStringOnProfileListItem(
+        position = 2,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "B"
+      )
+      scrollToPosition(position = 3)
+      matchStringOnProfileListItem(
+        position = 3,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "Ben"
+      )
+      scrollToPosition(position = 4)
+      matchStringOnProfileListItem(
+        position = 4,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "C"
+      )
+      scrollToPosition(position = 5)
+      matchStringOnProfileListItem(
+        position = 5,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "D"
+      )
+      scrollToPosition(position = 6)
+      matchStringOnProfileListItem(
+        position = 6,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "E"
+      )
+      scrollToPosition(position = 7)
+      matchStringOnProfileListItem(
+        position = 7,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "F"
+      )
+      scrollToPosition(position = 8)
+      matchStringOnProfileListItem(
+        position = 8,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "G"
+      )
+      scrollToPosition(position = 9)
+      matchStringOnProfileListItem(
+        position = 9,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "H"
       )
     }
   }
 
   @Test
-  // TODO(#973): Fix ProfileChooserFragmentTest
-  @Ignore
   fun testProfileChooserFragment_clickProfile_checkOpensPinPasswordActivity() {
     profileTestHelper.initializeProfiles()
     launch(ProfileChooserActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
       onView(atPosition(R.id.profile_recycler_view, 0)).perform(click())
       intended(hasComponent(PinPasswordActivity::class.java.name))
     }
   }
 
   @Test
-  // TODO(#973): Fix ProfileChooserFragmentTest
-  @Ignore
-  fun testProfileChooserFragment_clickAddProfile_checkOpensAdminAuthActivity_onBackButton_opensProfileChooserFragment() { // ktlint-disable max-line-length
```

Comment:
For this & next test: we shouldn't just be deleting them. Even though we can't test multi-intent flows in Robolectric, we should still be testing the navigation flows. Please re-add as needed and/or link to other tests for parts that are covered in other suites.

Reply:
Done

Reply:
Resolving on behalf of Ben